sdk/storage/Azure.Storage.Files.DataLake/samples/Sample01a_HelloWorld.cs

```diff
@@ -325,6 +326,113 @@ public void Read()
+                // For large files, it may be faster to call ReadTo()
+                #region Snippet:SampleSnippetDataLakeFileClient_ReadStreaming
+                Response<DataLakeFileReadStreamingResult> fileContents = file.ReadStreaming();
+                #endregion Snippet:SampleSnippetDataLakeFileClient_ReadStreaming
+                using (FileStream stream = File.OpenWrite(downloadPath))
+                {
+                    fileContents.Value.Content.CopyTo(stream);
+                }
+
+                // Verify the contents
+                Assert.AreEqual(SampleFileContent, File.ReadAllText(downloadPath));
+            }
+            finally
+            {
+                // Clean up after the test when we're finished
+                filesystem.Delete();
+            }
+        }
+
+        /// <summary>
+        /// Download a DataLake File content to a file.
+        /// </summary>
+        [Test]
+        public void ReadContent()
+        {
+            // Create a temporary Lorem Ipsum file on disk that we can upload
+            string originalPath = CreateTempFile(SampleFileContent);
+
+            // Get a temporary path on disk where we can download the file
+            string downloadPath = CreateTempPath();
+
+            // Make StorageSharedKeyCredential to pass to the serviceClient
+            string storageAccountName = StorageAccountName;
+            string storageAccountKey = StorageAccountKey;
+            Uri serviceUri = StorageAccountBlobUri;
+            StorageSharedKeyCredential sharedKeyCredential = new StorageSharedKeyCredential(storageAccountName, storageAccountKey);
+
+            // Create DataLakeServiceClient using StorageSharedKeyCredentials
+            DataLakeServiceClient serviceClient = new DataLakeServiceClient(serviceUri, sharedKeyCredential);
+
+            // Get a reference to a filesystem named "sample-filesystem-read" and then create it
+            DataLakeFileSystemClient filesystem = serviceClient.GetFileSystemClient("sample-filesystem-read");
+            filesystem.Create();
+            try
+            {
+                // Get a reference to a file named "sample-file" in a filesystem
+                DataLakeFileClient file = filesystem.GetFileClient("sample-file");
+
+                // First upload something the DataLake file so we have something to download
+                file.Upload(File.OpenRead(originalPath));
+
+                // Download the DataLake file's contents and save it to a file
+                // The ReadContentAsync() API downloads a file in a single requests.
+                // For large files, it may be faster to call ReadTo()
+                #region Snippet:SampleSnippetDataLakeFileClient_ReadContent
+                Response<DataLakeFileReadResult> fileContents = file.ReadContent();
+                #endregion Snippet:SampleSnippetDataLakeFileClient_ReadContent
```

Comment:
Same here, demonstrating expected typing of the primary field they'll be working with can be helpful.

```csharp
Response<DataLakeFileReadStreamingResult> fileContents = file.ReadStreaming();
BinaryData readData = fileContents.Value.Content;
```